### ISSUE TITLE
Configuration to Defer connection to Temporal on Application Start - Eg, using the Extension for the Client - 

### ISSUE THREAD

**Issue by akil-rails** [quarkiverse/quarkus-temporal#129]
I see that the extension connects to the temporal cloud at application start, this is great for workers, but not so good for the client.

I suppose, a common use case would be to use the WorkflowClient in a service layer (say a quarkus REST resource). The service would accept the request, and attempt a start of the workflow, and if it fails, it could decide to store it and try again. The service should be available to accept requests, irrespective of its connectivity to the temporal server, and could in-fact report that in the /q/health endpoint. 

A configuration option to do this would be great..

**Comment by melloware** [comment]
We had discussed this here: https://github.com/quarkiverse/quarkus-temporal/issues/70  cc @rmanibus ?

**Comment by akil-rails** [comment]
The extension already handles any network break & reconnection gracefully (once the application has started) .
A configuration option to let the application decide if the startup should fail if a connection isn't established will be very useful - specifically for temporal cloud uses, the network connectivity can be intermittent, and should not prevent application startups..

**Comment by melloware** [comment]
@akil-rails any interest in adding this feature?

**Comment by akil-rails** [comment]
Yes, though I need some guidance - I don't understand what part of the extension stops the app from starting ?

**Comment by melloware** [comment]
@rmanibus can provide some guidance here...
